bundles/org.openhab.binding.heliosventilation/src/main/java/org/openhab/binding/heliosventilation/internal/HeliosVentilationHandler.java

```diff
@@ -0,0 +1,458 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.io.transport.serial.PortInUseException;
+import org.eclipse.smarthome.io.transport.serial.SerialPort;
+import org.eclipse.smarthome.io.transport.serial.SerialPortEvent;
+import org.eclipse.smarthome.io.transport.serial.SerialPortEventListener;
+import org.eclipse.smarthome.io.transport.serial.SerialPortIdentifier;
+import org.eclipse.smarthome.io.transport.serial.SerialPortManager;
+import org.eclipse.smarthome.io.transport.serial.UnsupportedCommOperationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HeliosVentilationHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@NonNullByDefault
+public class HeliosVentilationHandler extends BaseThingHandler implements SerialPortEventListener {
+    private static final int BUSMEMBER_MAINBOARD = 0x11;
+    private static final int BUSMEMBER_SLAVEBOARDS = 0x10;
+    private static final byte BUSMEMBER_CONTROLBOARDS = (byte) 0x20;
+    private static final int BUSMEMBER_REC_MASK = 0xF0; // interpreting frames delivered to BUSMEMBER_ME &
+                                                        // BUSMEMBER_REC_MASK
+    private static final int BUSMEMBER_ME = 0x2F; // used as sender when communicating with the helios system
+    private static final int POLL_OFFLINE_THRESHOLD = 3;
+
+    private static final HashMap<Byte, HeliosVentilationDataPoint> DATAPOINTS;
+    private static final Logger LOGGER;
+    static {
+        /* logger is used by readChannelProperties() so we need to initialize logger first. */
+        LOGGER = LoggerFactory.getLogger(HeliosVentilationHandler.class);
+        DATAPOINTS = readChannelProperties();
+    }
+
+    private HeliosVentilationConfiguration config = new HeliosVentilationConfiguration();
+
+    private SerialPortManager serialPortManager;
+
+    private @Nullable SerialPort serialPort;
+    private @Nullable InputStream inputStream;
+    private @Nullable OutputStream outputStream;
+
+    private @Nullable ScheduledFuture<?> pollingTask;
+    private int pollCounter;
+
+    /**
+     * store received data for read-modify-write operations on bitlevel
+     */
+    private HashMap<Byte, Byte> memory = new HashMap<Byte, Byte>();
+
+    public HeliosVentilationHandler(Thing thing, final SerialPortManager serialPortManager) {
+        super(thing);
+        this.serialPortManager = serialPortManager;
+    }
+
+    /**
+     * parse datapoints from properties
+     *
+     * @return
+     */
+    private static HashMap<Byte, HeliosVentilationDataPoint> readChannelProperties() {
```

Comment:
I would suggest to move this to the `HeliosVentilationBindingConstants`, you can make the static logger a normal one then and it's constant anyway.

Reply:
Including DATAPOINTS you mean? I can move it for sure, but I don't get why that is better and whether I should move all constants (e. g. BUSMEMBER_MAINBOARD etc.) there. Also why should the Logger then be non-static? I just would need two: a static on in HeliosVentilationBindingConstants and a normal one in the handler class...